### PR TITLE
feat: register atomic contract

### PR DESCRIPTION
**Pull request by asiaziola**

### CODE REVIEW

File: src/gateway/router/routes/registerContractRoute.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  const request = async () => {
+    return fetch(`${BUNDLR_GATEWAY_URL}/tx/${id}/tags`).then((res) => {
+      return res.ok ? res.json() : Promise.reject(res);
+    });
+  };
+  try {
+    response = await backOff(request, {
+      delayFirstAttempt: false,
+      maxDelay: 2000,
+      numOfAttempts: 5,
```
Comment (asiaziola) [inline src/gateway/router/routes/registerContractRoute.ts:131]: not sure what values here would be appropriate

File: src/gateway/router/routes/registerContractRoute.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    const ownerAddress = txMetadata.address;
+    const contentType = tags.find((t: any) => t.name == 'Content-Type')!.value;
+    const testnet = getTestnetTag(tags);
+    const manifest = evalManifest(tags);
+
+    contractTx = {
+      id: txId,
+      owner: ownerAddress,
+      data: null,
+      signature: '',
```
Comment (asiaziola) [inline src/gateway/router/routes/registerContractRoute.ts:65]: waiting for bundlr response.. currently we dont have an option to retrieve the data item signature (unless we request arweave gateway)

Comment (asiaziola) [reply src/gateway/router/routes/registerContractRoute.ts]: done

File: tools/atomic.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      { name: 'Title', value: 'Asset' },
+      { name: 'Description', value: 'Description' },
+      { name: 'Type', value: 'Text' },
+    ];
+
+    const transaction = bundlr.createTransaction(data, { tags: contractTags });
+    await transaction.sign();
+    const tx = await transaction.upload();
+
+    await fetch(`http://localhost:5666/gateway/contracts/register`, {
```
Comment (ppedziwiatr) [inline tools/atomic.ts:46]: ?

File: src/gateway/router/routes/contractDataRoute.ts [hunk 4]
```diff
@@ -86,21 +89,29 @@ async function getContractData(
...
 
-  const tx = new Transaction({ ...JSON.parse(strData) });
-  const txData = Buffer.from(tx.data);
-
-  // get contract transaction content type from its tag
-  const contentType = getContentType(tx);
-  logger.debug(`Content type for id: ${id}: `, contentType);
-
-  return { data: txData, contentType };
+  if (deploymentType == 'warp-external') {
```
Comment (ppedziwiatr) [inline src/gateway/router/routes/contractDataRoute.ts:92]: extract to a type and describe its meaning.

File: src/gateway/router/routes/registerContractRoute.ts [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+      }
+    }
+  }`;
+
+export async function registerContractRoute(ctx: Router.RouterContext) {
+  const { logger, gatewayDb } = ctx;
+
+  let initStateRaw = '';
+  let contractTx = null;
+  let txId = '';
```
Comment (ppedziwiatr) [inline src/gateway/router/routes/registerContractRoute.ts:29]: add bundlr node as another query param (and validate!)

**Review by ppedziwiatr — CHANGES_REQUESTED** [review]

File: tools/atomic.ts (hunk 3 above)
Comment (asiaziola) [reply tools/atomic.ts]: done

File: src/gateway/router/routes/contractDataRoute.ts (hunk 4 above)
Comment (asiaziola) [reply src/gateway/router/routes/contractDataRoute.ts]: done

File: src/gateway/router/routes/registerContractRoute.ts (hunk 5 above)
Comment (asiaziola) [reply src/gateway/router/routes/registerContractRoute.ts:31]: done

File: src/gateway/router/routes/registerContractRoute.ts [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+      owner: ownerAddress,
+      type: type,
+      pst_ticker: type == 'pst' ? initState?.ticker : null,
+      pst_name: type == 'pst' ? initState?.name : null,
+      block_height: getCachedNetworkData().cachedNetworkInfo.height,
+      block_timestamp: getCachedNetworkData().cachedBlockInfo.timestamp,
+      content_type: contentType,
+      contract_tx: contractTx,
+      bundler_contract_tx_id: txId,
+      bundler_contract_node: BUNDLR_NODE2_URL,
```
Comment (ppedziwiatr) [inline src/gateway/router/routes/registerContractRoute.ts:96]: zjebałaś.

**Review by ppedziwiatr — CHANGES_REQUESTED** [review]

File: src/gateway/router/routes/registerContractRoute.ts (hunk 6 above)
Comment (asiaziola) [reply src/gateway/router/routes/registerContractRoute.ts]: weź.

**Pull request merged**